Bottom: d45c7bb9a2a0b39141c8ca91b655e106ef2366ca
Top:    ee068471f46f3964bf29f04d3a23f1cf03d00a7c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 02:10:04 +0100

Try optimising polygon operations by keeping an rtree with their contours

InsertHoles() already creates an rtree of contours to do its internal
processing, and it is possible that intersect() could be sped up by
its existance as well.




---

diff --git a/src/polyarea.h b/src/polyarea.h
index 42acc1a..819b074 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -128,6 +128,7 @@ struct POLYAREA
 {
     POLYAREA *f, *b;
     PLINE *contours;
+    void *contour_tree;
 };
 
 BOOLp poly_M_Copy0(POLYAREA ** dst, const POLYAREA * srcfst);
